gufe/components/solventcomponent.py

```diff
@@ -22,39 +56,94 @@ class SolventComponent(Component):
+                 num_solvent: Optional[int] = None,
                  positive_ion: str = 'Na+',
                  negative_ion: str = 'Cl-',
                  neutralize: bool = True,
-                 ion_concentration: unit.Quantity = 0.15 * unit.molar):
+                 ion_concentration: Optional[unit.Quantity] = 0.15 * unit.molar,
+                 solvent_padding: Optional[unit.Quantity] = 1.2 * unit.nanometer,
+                 solvent_density: Optional[unit.Quantity] = None,
+                 box_shape: Optional[Literal['cube', 'dodecahedron', 'octahedron']] = 'cube',
+                 box_vectors: Optional[unit.Quantity] = None,):
         """
         Parameters
         ----------
-        smiles : str, optional
+        smiles : str
           smiles of the solvent, default 'O' (water)
+        num_solvent : int, optional
+          number of solvent molecules present, if ``None``, will be determined
+          by either `solvent_density` if defined, or the utilized solvation
+          backend (Protocol specific). Cannot be defined as the same time
+          as both `solvent_density` and either one of `solvent_padding` or
+          `box_vectors`. Default `None`
         positive_ion, negative_ion : str
           the pair of ions which is used to neutralize (if neutralize=True) and
           bring the solvent to the required ionic concentration.  Must be a
           positive and negative monoatomic ions, defaults "Na+", "Cl-"
-        neutralize : bool, optional
+        neutralize : bool
           if the net charge on the chemical state is neutralized by the ions in
           this solvent component.  Default `True`
-        ion_concentration : openff-units.unit.Quantity, optional
+        ion_concentration : openff.units.unit.Quantity, optional
           ionic concentration required, default 0.15 * unit.molar
           this must be supplied with units, e.g. "1.5 * unit.molar"
+        solvent_padding : openff.units.unit.Quantity, optional
+          padding distance away from solute to use in constructing the
+          solvent box. Must be supplied with units, e.g.
+          "1.2 * unit.nanometer". Cannot be defined at the same time as
+          `box_vectors`. Cannot be defined alongside both `num_solvent`
+          and `solvent_density`. Default 1.2 * unit.nanometer
+        solvent_density : openff.units.unit.Quantity, optional
+          Target density of solvated systems. Must be defined with units
+          compatible with g / mL, e.g. "850 * unit.kilogram / unit.meter**3".
+          Cannot be defined alongside `num_solvent` if either `box_vector`
+          or `solvent_padding` are defined. Default `None`
+        box_shape : str, optional
+          Defined the shape of the solvent box being built. Can be one of
+          'cube', 'dodecahedron', and 'octahedron'. Cannot be defined alongside
+          `box_vectors`. Default 'cube'
+        box_vectors : openff.units.unit.Quantity, optional
```

Comment:
Putting in `box_vectors` will be needed for explicit solvents, but I don't think we'll need to backport that field to this definition